### PR TITLE
chore(deps): Update `@guardian/cdk` to v41.0.1

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Updates `@guardian/cdk` from 34.0.0 to 41.0.1. This is quite a big change - I've tried to annotate the snapshot changes with explanations.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -27,7 +27,8 @@ lazy val root = (project in file("."))
     riffRaffArtifactResources  := Seq(
       riffRaffPackageType.value -> s"${name.value}/${name.value}.deb",
       baseDirectory.value / "riff-raff.yaml" -> "riff-raff.yaml",
-      baseDirectory.value / "cdk/cdk.out/PrismEc2App.template.json" -> "cloudformation/PrismEc2App.template.json"
+      baseDirectory.value / "cdk/cdk.out/Prism-CODE.template.json" -> "cloudformation/Prism-CODE.template.json",
+      baseDirectory.value / "cdk/cdk.out/Prism-PROD.template.json" -> "cloudformation/Prism-PROD.template.json"
```
Comment (akash1810) [inline build.sbt:31]: From https://github.com/guardian/cdk/releases/tag/v39.0.0.

File: cdk/lib/__snapshots__/prism-access.test.ts.snap [hunk 2]
```diff
@@ -105,9 +97,7 @@ Object {
           },
           Object {
             "Key": "Stage",
-            "Value": Object {
-              "Ref": "Stage",
-            },
+            "Value": "INFRA",
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-access.test.ts.snap:100]: This is the (singleton) stack set, so taking the opportunity to move to the `INFRA` stage.

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 3]
```diff
@@ -2,22 +2,7 @@
 
 exports[`The PrismEc2App stack matches the snapshot 1`] = `
 Object {
-  "Mappings": Object {
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:5]: This snapshot represents PROD only now.

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 4]
```diff
@@ -813,11 +746,30 @@ dpkg -i /prism/prism.deb",
...
+                    ":parameter/PROD/deploy/prism",
+                  ],
+                ],
+              },
+            },
+            Object {
+              "Action": Array [
+                "ssm:GetParameters",
+                "ssm:GetParameter",
+              ],
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:758]: From https://github.com/guardian/cdk/releases/tag/v34.1.0.

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 5]
```diff
@@ -1028,10 +960,10 @@ dpkg -i /prism/prism.deb",
             },
           },
         ],
-        "EvaluationPeriods": 12,
+        "EvaluationPeriods": 60,
         "MetricName": "UnHealthyHostCount",
         "Namespace": "AWS/ApplicationELB",
-        "Period": 300,
+        "Period": 60,
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:966]: From https://github.com/guardian/cdk/releases/tag/v34.0.1.

File: cdk/lib/prism-ec2-app.ts [hunk 6]
```diff
@@ -32,14 +38,16 @@ export class PrismEc2App extends GuStack {
...
-        PROD: { domainName: "prism.gutools.co.uk" },
-      },
-      monitoringConfiguration: {
-        snsTopicName: "devx-alerts",
-        http5xxAlarm: false,
-        unhealthyInstancesAlarm: true,
+        domainName: props.domainName,
       },
+      monitoringConfiguration:
+        this.stage === "PROD"
```
Comment (akash1810) [inline cdk/lib/prism-ec2-app.ts:44]: Only monitor and alarm on PROD.

File: riff-raff.yaml [hunk 7]
```diff
@@ -5,7 +5,9 @@ deployments:
     type: cloud-formation
     app: prism
     parameters:
-      templatePath: PrismEc2App.template.json
+      templateStagePaths:
+        CODE: Prism-CODE.template.json
+        PROD: Prism-PROD.template.json
```
Comment (akash1810) [inline riff-raff.yaml:10]: From https://github.com/guardian/cdk/releases/tag/v39.0.0.

**Comment by akash1810** [comment]
The application stack has been successfully deployed to CODE (via Riff-Raff). The stack set has been successfully deployed (manually).

**Review by jacobwinch — APPROVED** [review]

**Pull request merged**